ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/RemoveUnnecessaryArgumentFix.fs

```diff
@@ -0,0 +1,33 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Psi.Tree
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Util
+open System.Linq
+
+type RemoveUnnecessaryArgumentFix(notAFunctionExpr: ISynExpr) =
+    inherit FSharpQuickFixBase()
+    
+    let removeWhitespacesBetweenArgs (prefixApp: IPrefixAppExpr) =
+        let whitespaces = prefixApp.Children().Where(fun x -> x.IsWhitespaceToken())
+        for wp in whitespaces do ModificationUtil.DeleteChild(wp)
+    
+    new (warning: NotAFunctionError) =
+        RemoveUnnecessaryArgumentFix(warning.NotAFunctionExpr)
+    
+    override x.Text = "Remove unnecessary arguments"
+
+    override x.IsAvailable _ =
+        isValid notAFunctionExpr &&
+        match notAFunctionExpr with
+        | :? ILiteralExpr -> false
+        | _ -> true
+
+    override x.ExecutePsiTransaction _ =
+        use writeCookie = WriteLockCookie.Create(notAFunctionExpr.IsPhysical())
+        
+        let argsToRemove = getUnexpectedArgsWithVisitor removeWhitespacesBetweenArgs notAFunctionExpr       
+        for arg in argsToRemove do ModificationUtil.DeleteChild(arg)

```

Comment:
Try working with argument tree ranges instead. It'd significantly reduce tree modifications count, which in turn would prevent lots of additional things; like formatting changed nodes, from happening.